cmd/observer/node_builder/observer_builder.go

```diff
@@ -0,0 +1,878 @@
+	logTxTimeToFinalized         bool
+	logTxTimeToExecuted          bool
+	logTxTimeToFinalizedExecuted bool
+	retryEnabled                 bool
+	rpcMetricsEnabled            bool
+	baseOptions                  []cmd.Option
+
+	PublicNetworkConfig PublicNetworkConfig
+}
+
+type PublicNetworkConfig struct {
+	// NetworkKey crypto.PublicKey // TODO: do we need a different key for the public network?
+	BindAddress string
+	Network     network.Network
+	Metrics     module.NetworkMetrics
+}
+
+// DefaultObserverServiceConfig defines all the default values for the ObserverServiceConfig
+func DefaultObserverServiceConfig() *ObserverServiceConfig {
+	return &ObserverServiceConfig{
+		collectionGRPCPort: 9000,
+		executionGRPCPort:  9000,
+		rpcConf: rpc.Config{
+			UnsecureGRPCListenAddr:    "0.0.0.0:9000",
+			SecureGRPCListenAddr:      "0.0.0.0:9001",
+			HTTPListenAddr:            "0.0.0.0:8000",
+			RESTListenAddr:            "",
+			CollectionAddr:            "",
+			HistoricalAccessAddrs:     "",
+			CollectionClientTimeout:   3 * time.Second,
+			ExecutionClientTimeout:    3 * time.Second,
+			MaxHeightRange:            backend.DefaultMaxHeightRange,
+			PreferredExecutionNodeIDs: nil,
+			FixedExecutionNodeIDs:     nil,
+		},
+		ExecutionNodeAddress:         "localhost:9000",
+		logTxTimeToFinalized:         false,
+		logTxTimeToExecuted:          false,
+		logTxTimeToFinalizedExecuted: false,
+		retryEnabled:                 false,
+		rpcMetricsEnabled:            false,
+		apiRatelimits:                nil,
+		apiBurstlimits:               nil,
+		staked:                       false, // deprecated but kept to support temporary boostrap code
+		bootstrapNodeAddresses:       []string{},
+		bootstrapNodePublicKeys:      []string{},
+		supportsPublicFollower:       false,
+		PublicNetworkConfig: PublicNetworkConfig{
+			BindAddress: cmd.NotSet,
+			Metrics:     metrics.NewNoopCollector(),
+		},
+		observerNetworkingKeyPath: cmd.NotSet,
+	}
+}
+
+// ObserverServiceBuilder provides the common functionality needed to bootstrap a Flow staked and observer
+// It is composed of the FlowNodeBuilder, the ObserverServiceConfig and contains all the components and modules needed for the
+// staked and observers
+type ObserverServiceBuilder struct {
+	*cmd.FlowNodeBuilder
+	*ObserverServiceConfig
+
+	// components
+	LibP2PNode                 *p2p.Node
+	FollowerState              protocol.MutableState
+	SyncCore                   *synchronization.Core
+	RpcEng                     *rpc.Engine
+	FinalizationDistributor    *pubsub.FinalizationDistributor
+	FinalizedHeader            *synceng.FinalizedHeaderCache
+	CollectionRPC              access.AccessAPIClient
+	TransactionTimings         *stdmap.TransactionTimings
+	CollectionsToMarkFinalized *stdmap.Times
+	CollectionsToMarkExecuted  *stdmap.Times
+	BlocksToMarkExecuted       *stdmap.Times
+	TransactionMetrics         module.TransactionMetrics
+	PingMetrics                module.PingMetrics
```

Comment:
I don't think any of these are needed for observers

Reply:
Fixed: https://github.com/GetElastech/flow-go/commit/d6bc312671ee2e513600843a07034800a3fa3c6a

Reply:
Fixed by three different changes.
Fixed: https://github.com/GetElastech/flow-go/commit/d4ff7a6d2221a5a6ec985e5ff19b6959f124c6a8
Fixed: https://github.com/GetElastech/flow-go/commit/0ea1f2de8e0aa7cac026568cd515b1be0e0e48c8
Fixed: https://github.com/GetElastech/flow-go/commit/b558c135a1c6516b298a383c819e841b8c4106d1